analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_dryspells.md

```diff
@@ -0,0 +1,827 @@
+df_for_sel=df_for[(df_for.ADM1_EN.isin(sel_adm))&(df_for.date.dt.month.isin(sel_months))&(df_for.leadtime.isin(sel_leadtime))&(df_for.season_approx.isin(seas_years))]
+```
+
+### Load observational data
+
+```python
+#load the monthly precipitation data
+df_obs_month=pd.read_csv(monthly_precip_path,parse_dates=["date"])
+df_obs_month["date_month"]=df_obs_month.date.dt.to_period("M")
+df_obs_month["season_approx"]=np.where(df_obs_month.date.dt.month>=10,df_obs_month.date.dt.year,df_obs_month.date.dt.year-1)
+
+#select relevant admins and months
+df_obs_month_sel=df_obs_month[(df_obs_month.ADM1_EN.isin(sel_adm))&(df_obs_month.date.dt.month.isin(sel_months))&(df_obs_month.season_approx.isin(seas_years))]
+```
+
+```python
+df_ds=load_dryspell_data(all_dry_spells_list_path)
+```
+
+```python
+#check that same number of months in observed and forecasted
+print("number of months in forecasted data",len(df_for_sel.date.unique()))
+print("number of months in observed data",len(df_obs_month_sel.date.unique()))
+```
+
+### set threshold based on x% probability
+
+```python
+probability=0.5
+```
+
+```python
+#compute the value for which x% of members forecasts the precipitation to be below or equal to that value
+df_for_quant=df_for_sel.groupby(["date","ADM1_EN","leadtime"],as_index=False).quantile(probability)
+df_for_quant["date_month"]=df_for_quant.date.dt.to_period("M")
+```
+
+```python
+df_for_quant[["date","leadtime","mean_cell"]].head()
+```
+
+```python
+#plot the distribution of precipitation. Can clearly see that for leadtime=1 the values are more spread (but leadtime=1 is really the month that is currently occurring so less useful)
+g = sns.FacetGrid(df_for_quant, height=5, col="leadtime",col_wrap=3)
+g.map_dataframe(sns.histplot, "mean_cell",common_norm=False,kde=True,alpha=1,binwidth=10)
+
+for ax in g.axes.flatten():
+    ax.tick_params(labelbottom=True)
+    ax.set_ylabel("Number of months")
+    ax.set_xlabel("Total monthly precipitation (mm)")
+```
+
+```python
+#include all dates present in the observed rainfall df but not in the dry spell list, i.e. where no dryspells were observed, by merging outer
+df_ds_for=df_ds.merge(df_for_quant,how="right",on=["ADM1_EN","date_month"])
+df_ds_for.loc[:,"dry_spell"]=df_ds_for.dry_spell.replace(np.nan,0).astype(int)
+```
+
+```python
+df_ds_for["month"]=df_ds_for.date_month.dt.month
+df_ds_for["month_name"]=df_ds_for.month.apply(lambda x: calendar.month_name[x])
+df_ds_for["month_abbr"]=df_ds_for.month.apply(lambda x: calendar.month_abbr[x])
+df_ds_for_labels=df_ds_for.replace({"dry_spell":{0:"no",1:"yes"}}).sort_values("dry_spell",ascending=True)
+```
+
+```python
+#for some reason facetgrid doesn't want to show the values if there is only one occurence (i.e. in January..)
+g = sns.FacetGrid(df_ds_for_labels, height=5, col="leadtime",row="month_name",hue="dry_spell",palette={"no":"#CCE5F9","yes":'#F2645A'})
```

Comment:
It's not super important but it might make your life easier to make variables or a dictionary of the colours that you like to use

Reply:
Colors thanks to the [HDX data viz guidelines](https://company-200741.frontify.com/d/1GC9otcJfhPe/visual-identity) :)